packages/react-composites/src/composites/CallComposite/adapter/OnFetchProfileCallback.ts

```diff
@@ -48,10 +44,7 @@ export const createProfileStateModifier = (
         if (profile?.displayName && originalParticipants[key].displayName !== profile?.displayName) {
           cachedDisplayName[key] = profile?.displayName;
         }
-        shouldNotifyUpdates = true;
       }
-      // notify update only when there is a change, which most likely will trigger modifier and setState again
-      shouldNotifyUpdates && notifyUpdate();
```

Comment:
@PorterNan do you know why we notify an update here? This caused an infinite cycle when I set onFetchProfile on the CallAdapterOptions

Reply:
Cause displayName could be just undefined at the very beginning then becomes available later
The workflow is like
async getDisplayName -> displayName still undefined -> render component -> finished async call and get displayName and render it, we need to notify update to adapter to tell it the avatar state has been changed

Reply:
We need to ensure displayName is always the same from the Onfetch functions (and the key is not undefined I guess?), let's have an offline session together to debug this one

Reply:
Thanks for the clarification. I reverted the changes here and the infinite loop I was seeing is no longer happening. My mistake.

Reply:
I see . Thanks for the clarification. Turns out, I only needed to set the `shouldNotifyUpdates` flag to true when the display name is different from the cached display name.